mythx_cli/analyze/solidity.py

```diff
@@ -191,48 +199,55 @@ def generate_payloads(
+        else:
+            # use solc for compilation
+            self.setup_solcx(solc_version)
+            try:
+                cwd = str(Path.cwd().absolute())
+                LOGGER.debug(f"Compiling {self.target} under allowed path {cwd}")
+                result = self.solcx_compile(cwd, remappings, enable_scribble)
+            except solcx.exceptions.SolcError as e:
+                raise click.exceptions.UsageError(
+                    f"Error compiling source with solc {solc_version}: {e}"
                 )
-                click.echo("=====STDERR=====")
-                click.echo(process.stderr.decode())
-                click.echo("=====STDOUT=====")
-                process.stdout.decode()
-                sys.exit(process.returncode)
-
-            # don't delete temp file on close but manually unlink
-            # after payload has been generated
-            scribble_output_f = tempfile.NamedTemporaryFile(
-                mode="w+", delete=False, suffix=".sol"
-            )
-            scribble_stdout = process.stdout.decode()
-            scribble_output_f.write(scribble_stdout)
-            scribble_file = scribble_output_f.name
-            scribble_output_f.close()
-
-        try:
-            cwd = str(Path.cwd().absolute())
-            LOGGER.debug(
-                f"Compiling {scribble_file or self.target} under allowed path {cwd}"
-            )
-            result = self.solcx_compile(cwd, remappings, enable_scribble, scribble_file)
-        except solcx.exceptions.SolcError as e:
-            raise click.exceptions.UsageError(
-                f"Error compiling source with solc {solc_version}: {e}"
-            )
 
-        payload = self.payload_from_sources(result, scribble_file, solc_version)
+        # instrument with scribble if requested
+        # scribble_file = None
+        # if enable_scribble:
+        #     process = subprocess.run(
+        #         [scribble_path, self.target],
+        #         stdout=subprocess.PIPE,
+        #         stderr=subprocess.PIPE,
+        #     )
+        #     if process.returncode != 0:
+        #         click.echo(
+        #             f"Scribble has encountered an error (code: {process.returncode})"
+        #         )
+        #         click.echo("=====STDERR=====")
+        #         click.echo(process.stderr.decode())
+        #         click.echo("=====STDOUT=====")
+        #         process.stdout.decode()
+        #         sys.exit(process.returncode)
+        #
+        #     # don't delete temp file on close but manually unlink
+        #     # after payload has been generated
+        #     scribble_output_f = tempfile.NamedTemporaryFile(
+        #         mode="w+", delete=False, suffix=".sol"
+        #     )
+        #     scribble_stdout = process.stdout.decode()
+        #     scribble_output_f.write(scribble_stdout)
+        #     scribble_file = scribble_output_f.name
+        #     scribble_output_f.close()
```

Comment:
Should this commented section be included?

Reply:
Fixed in 33bf358